src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -73,6 +75,49 @@ class RestrictedPatientsService(
       .let { addRestrictedPatient(it) }
   }
 
+  @Transactional
+  fun prisonerReleased(offenderNo: String) {
+    if (restrictedPatientsRepository.existsById(offenderNo)) {
+      log.info("Ignoring release of prisoner $offenderNo as they are already a restricted patient")
+      return
+    }
+
+    val lastMovement = getReleaseToHospitalMovement(offenderNo)
+    if (lastMovement == null) {
+      log.info("Ignoring release of prisoner $offenderNo as they are not being released to a hospital")
```

Comment:
perhaps debug as this would be vast majority

Reply:
But debug also goes to App Insights... maybe just drop this entirely then.